src/main/java/com/cloudbees/hudson/plugins/folder/ChildNameGenerator.java

```diff
@@ -99,58 +85,6 @@ public abstract class ChildNameGenerator<P extends AbstractFolder<I>, I extends
      */
     public static final String CHILD_NAME_FILE = "name-utf8.txt";
 
-    private static final Map<Trace,String> idealNames = new WeakHashMap<>();
-
-    /**
-     * Work-around helper method to "fix" {@link Item} constructors that have on-disk side-effects and therefore
-     * need {@link Item#getRootDir()} to work during the constructor.
-     * @param project the {@link AbstractFolder}.
-     * @param itemName the name that will be returned by {@link Item#getName()} when the item is constructed. This is
-     *                 the second parameter of {@link AbstractItem#AbstractItem(ItemGroup, String)}. This one would be
-     *                 the one with URL path segment escaping.
-     * @param idealName the original name before whatever URL path segment escaping you applied
-     * @return the {@link Trace} to keep track of when we can remove the memory of the creation process. Please
-     * {@link Trace#close()} the trace after the item is created.
-     */
-    @NonNull
-    public static Trace beforeCreateItem(@NonNull AbstractFolder<?> project,
```

Comment:
https://github.com/jenkinsci/branch-api-plugin/blob/43d8bbd8b2c7ac526faf5d047af5b8e783cef681/src/main/java/jenkins/branch/OrganizationFolder.java#L1440 would need to be removed I suppose?

Reply:
> would need to be

https://github.com/jenkinsci/branch-api-plugin/pull/506